### ISSUE TITLE
KI-HS50-H

### ISSUE THREAD

**Issue by Cullingsong** [scottyphillips/echonetlite_homeassistant#108]
Hi - I noticed you added the KI-HS70 which is great. I have the KI-HS50 and it seems to mostly work.  The power reporting doesn't quite work right, and the air flow setting doesn't seem to do anything.  Would it be possible to see if you can add support for this device?

Thanks!

**Comment by scottyphillips** [comment]
Airflow settings unfortunately are trial and error; have you tried every permutation in the select option?

Are you able to clarify what you mean by “power reporting doesn’t quite work right?”

can you please run a debug and post the log output as well as any relevant screenshots?

**Comment by Cullingsong** [comment]
Hi - thanks for getting back so quickly. Is there a better way to get debug other than 'logger' in the config file?

Are you able to clarify what you mean by “power reporting doesn’t quite work right?”  --> Example: Turning it 'off' works, but the switch changes back to 'on' after about 2 seconds. The device is still off though.

**Comment by scottyphillips** [comment]
Are talking about the switch in the fan entity or are you talking about a switch entity?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Cullingsong** [comment]
Switch in the fan entity 

![image](https://user-images.githubusercontent.com/16530444/184090231-ee85ae7f-2fbf-4c3b-84ba-604a2ae3db64.png)

**Comment by nao-pon** [comment]
Would changing the following change the result?
HA must be restarted after the change.

fan.py L.98 - 
```python
    async def async_turn_on(
        self,
        speed: str = None,
        percentage: int = None,
        preset_mode: str = None,
        **kwargs,
    ):
        """Turn on."""
        if await self._connector._instance.on():
            self.async_write_ha_state()

    async def async_turn_off(self):
        """Turn off."""
        if await self._connector._instance.off():
            self.async_write_ha_state()
```

**Comment by Cullingsong** [comment]
Hi nao-pon,

I just changed it but it's still acting the same. I'm just about to head out of town for a week. I'll troubleshoot it a bit more when I get back.

Thanks

**Comment by scottyphillips** [comment]
Are you using the current HACS release or direct from GitHub? I’ve still got the most up to date code with all of the @nao-pon updates tagged as prerelease..

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Cullingsong** [comment]
Hi - sorry I'm back now.  I'm using the HACs version.  I'll give the git-hub version a try shortly

**Comment by nao-pon** [comment]
This issue may have been resolved in version 3.6.1. Please try 3.6.1.

**Comment by nao-pon** [comment]
@Cullingsong Version 3.6.2 has been released. It contains a modification that provides a more stable response.
